src/main/java/gyro/google/storage/BucketResource.java

```diff
@@ -0,0 +1,651 @@
+                            this,
+                            "labels",
+                            String.format("Invalid key/value => '%s:%s'. Keys and values must be less than 64 characters "
+                                   + "and contain only lowercase letters, numeric characters, international characters, "
+                                   + "underscores, and dashes. Keys must start with letter or international character and "
+                                   + "must not be empty.", key, value)));
+                }
+            }
+        }
+
+        return errors;
+    }
+
+    @Override
+    public void copyFrom(Bucket model) {
+        setId(model.getId());
+        setName(model.getName());
+        setLabels(model.getLabels());
+        setLocation(model.getLocation());
+        setStorageClass(model.getStorageClass());
+        setSelfLink(model.getSelfLink());
+
+        if (model.getBilling() != null) {
+            BucketBilling bucketBilling = newSubresource(BucketBilling.class);
+            bucketBilling.copyFrom(model.getBilling());
+            setBilling(bucketBilling);
+        }
+
+        if (model.getEncryption() != null) {
+            BucketEncryption bucketEncryption = newSubresource(BucketEncryption.class);
+            bucketEncryption.copyFrom(model.getEncryption());
+            setEncryption(bucketEncryption);
+        }
+
+        if (model.getIamConfiguration() != null) {
+            BucketIamConfiguration bucketIamConfiguration = newSubresource(BucketIamConfiguration.class);
+            bucketIamConfiguration.copyFrom(model.getIamConfiguration());
+            setIamConfiguration(bucketIamConfiguration);
+        }
+
+        if (model.getLifecycle() != null) {
+            BucketLifecycle bucketLifecycle = newSubresource(BucketLifecycle.class);
+            bucketLifecycle.copyFrom(model.getLifecycle());
+            setLifecycle(bucketLifecycle);
+        }
+
+        if (model.getLogging() != null) {
+            BucketLogging bucketLogging = newSubresource(BucketLogging.class);
+            bucketLogging.copyFrom(model.getLogging());
+            setLogging(bucketLogging);
+        }
+
+        if (model.getRetentionPolicy() != null) {
+            BucketRetentionPolicy policy = newSubresource(BucketRetentionPolicy.class);
+            policy.copyFrom(model.getRetentionPolicy());
+            setRetentionPolicy(policy);
+        }
+
+        if (model.getVersioning() != null) {
+            BucketVersioning bucketVersioning = newSubresource(BucketVersioning.class);
+            bucketVersioning.copyFrom(model.getVersioning());
+            setVersioning(bucketVersioning);
+        }
+
+        if (model.getWebsite() != null) {
+            BucketWebsite bucketWebsite = newSubresource(BucketWebsite.class);
+            bucketWebsite.copyFrom(model.getWebsite());
+            setWebsite(bucketWebsite);
+        }
+
+        if (model.getCors() != null) {
```

Comment:
In case cors are emptied out
```suggestion
        getCors().clear();
        if (model.getCors() != null) {
```

Reply:
Fixed this along with other sub-resources.